Assets/Scripts/Utils/SteamManager.cs

```diff
@@ -164,16 +198,42 @@ private void OnlobbyCreated(LobbyCreated_t callback)
             return;
         NetworkManager.singleton.StartHost();
         IsHosting = true;
-        SteamMatchmaking.SetLobbyData(new CSteamID(callback.m_ulSteamIDLobby), hostkey, SteamUser.GetSteamID().ToString());
-        SteamMatchmaking.SetLobbyData(new CSteamID(callback.m_ulSteamIDLobby), "name", UserName);
+
+        lobbyID = new CSteamID(callback.m_ulSteamIDLobby);
+        SteamMatchmaking.SetLobbyData(lobbyID, hostkey, SteamUser.GetSteamID().ToString());
+        SteamMatchmaking.SetLobbyData(lobbyID, Lobby.NameProperty, UserName);
+        SteamMatchmaking.SetLobbyData(lobbyID, Lobby.AvailableSlotsProperty, Peer2PeerTransport.NumAvailableSlots.ToString());
+
+        // Update filterable information when necessary
+        ((Peer2PeerTransport)NetworkManager.singleton).OnPlayerReceived += UpdateAvailableSlots;
+        ((Peer2PeerTransport)NetworkManager.singleton).OnPlayerRemoved += UpdateAvailableSlots;
+        ((Peer2PeerTransport)NetworkManager.singleton).OnMatchStart += SetAsNotJoinable;
+        ((Peer2PeerTransport)NetworkManager.singleton).OnMatchEnd += SetAsJoinable;
+    }
+
+    private void UpdateAvailableSlots(PlayerDetails player)
+    {
+        SteamMatchmaking.SetLobbyData(lobbyID, Lobby.AvailableSlotsProperty, Peer2PeerTransport.NumAvailableSlots.ToString());
+    }
+
+    // TODO We wanna do this differently for rejoining disconnected player
+    private void SetAsNotJoinable()
+    {
+        SteamMatchmaking.SetLobbyJoinable(lobbyID, false);
+    }
+
+    private void SetAsJoinable()
+    {
+        SteamMatchmaking.SetLobbyJoinable(lobbyID, true);
     }
 
     private void OnJoinRequest(GameLobbyJoinRequested_t callback)
     {
         // TODO verify that the lobby *should* be joined by more players!
         //      and verify that this is run on the server!
+        Debug.Log("LOBBY JOIN REQUEST");
```

Comment:
I think we can remove this, also isn't the TODO now technically done? 🤔

Reply:
Seems to work now, though I'm not sure if this method does *exactly* what I think it does.